api/server/structs/conversions.go

```diff
@@ -1074,3 +1075,20 @@ func sszBytesToUint256String(b []byte) (string, error) {
 	}
 	return bi.String(), nil
 }
+
+func DepositSnapshotFromConsensus(ds *eth.DepositSnapshot) *DepositSnapshot {
+	finalized := make([]string, 0, len(ds.Finalized))
+	for i, f := range ds.Finalized {
+		if i >= depositsnapshot.DepositContractDepth {
```

Comment:
This function should not have logic in it, it should purely convert between types. You should check the length of the returned value inside the handler. Although I find it peculiar that we add constraints to a *returned* value, I guess it might be OK as an indication that we have a bug somewhere in the deposit snapshot code.

Reply:
You are right, I have modified it to check ahead of time as if the length of finalized exceeds the deposit contract depth we likely have an invalid snapshot